collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1236,7 +1238,7 @@ public void onClick(View v) {
                 }
 
                 if (mBackButton.isShown() && mNextButton.isShown()) {
-                    mBackButton.setEnabled(true);
+                    mBackButton.setEnabled(!firstQuestion);
```

Comment:
To many changes and to many places. I mentioned you don't need additional variable all you need you can add just here. You can check here the value of previous screen event and set enabled to true or false.